token/services/network/fabric/endorsement/approval.go

```diff
@@ -101,8 +93,21 @@ func (r *RequestApprovalView) Call(context view.Context) (interface{}, error) {
 	return tx.Envelope()
 }
 
+type Translator interface {
```

Comment:
I don't know how intuitive it is not to pass the RWSet as a param and include it in the fields of the translator. I expect the translator to be a service and act on different RWSets.